composer/core/precision.py

```diff
@@ -67,11 +69,11 @@ def get_precision_context(precision: Union[str, Precision]) -> Generator[None, N
         if te_installed and torch.cuda.get_device_capability()[0] > 8:
             from transformer_engine.common.recipe import DelayedScaling, Format
 
-            # These default values for fp8_recipe are taken from NVidia's docs. We may want to change
-            # these once we get a chance to do more convergence experiments.
-            # https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/examples/fp8_primer.html#id1
-            fp8_format = Format.HYBRID  # E4M3 during forward pass, E5M2 during backward pass
-            fp8_recipe = DelayedScaling(fp8_format=fp8_format, amax_history_len=16, amax_compute_algo='max')
+            if precision_config is None:
+                precision_config = {}
+            if 'fp8_format' in precision_config and isinstance(precision_config['fp8_format'], str):
```

Comment:
If it's none, let us init it to previous defaults. Update the docs for the same.

Reply:
Do you want to let the `DelayedScaling` class decide the defaults? Previously it looks like we defaulted the arguments to the default ones from NVIDIA's docs: https://github.com/mosaicml/composer/blob/b73c776725c05fbd5b7cf43b63a8ca85ed455af0/composer/core/precision.py#L70-L72

Right now it looks like the default parameters for the class match what we had before, except for `amax_history_len`, which is `1024` instead of `16`: https://github.com/NVIDIA/TransformerEngine/blob/6f7853dc9fb5cf4254760ec3994d5a09946aaa81/transformer_engine/common/recipe.py#L127-L129

So it depends on if we agree with the default values TE has already ^

Reply:
This is fine. I misunderstood.